### ISSUE TITLE
What's up with ` vue/experimental-script-setup-vars`?

### ISSUE THREAD

**Issue by FloEdelmann** [vuejs/eslint-plugin-vue#1882]
The [`vue/experimental-script-setup-vars`](https://eslint.vuejs.org/rules/experimental-script-setup-vars.html) rule was deprecated in #1528 (first released in [v7.13.0](https://github.com/vuejs/eslint-plugin-vue/releases/tag/v7.13.0)), but without any explanation or replacement rule.

It seems like [`vue/script-setup-uses-vars`](https://eslint.vuejs.org/rules/script-setup-uses-vars.html) (added in #1529, will be deprecated in #1849) is actually the replacement rule, as it was also first released in [v7.13.0](https://github.com/vuejs/eslint-plugin-vue/releases/tag/v7.13.0).

However, in [tests/lib/script-setup.js](https://github.com/vuejs/eslint-plugin-vue/blob/master/tests/lib/script-setup.js), `vue/experimental-script-setup-vars` is still hardcoded. When I try to replace it with `vue/script-setup-uses-vars` there, the test fails:

```diff
  1) script-setup test cases
       temporary supports.
         should not be marked.:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:

+ expected
- actual

+[]
-[
-  {
-    "column": 37
-    "endColumn": 42
-    "endLine": 5
-    "line": 5
-    "message": "'props' is not defined."
-    "messageId": "undef"
-    "nodeType": "Identifier"
-    "ruleId": "no-undef"
-    "severity": 2
-  }
-  {
-    "column": 7
-    "endColumn": 11
-    "endLine": 6
-    "line": 6
-    "message": "'emit' is not defined."
-    "messageId": "undef"
-    "nodeType": "Identifier"
-    "ruleId": "no-undef"
-    "severity": 2
-  }
-]
```

CC @ota-meshi

**Comment by FloEdelmann** [comment]
It seems like the syntax in the test is an early version of the `<script setup>` RFC:

https://github.com/vuejs/eslint-plugin-vue/blob/9344af8114d457eab996f3ec2439839a37b291fa/tests/lib/script-setup.js?rgh-link-date=2022-05-07T12%3A56%3A48Z#L39-L44

https://vuejs.org/api/sfc-script-setup.html has no references to `<script setup="props">`.

Since `vue/script-setup-uses-vars` has [its own tests](https://github.com/vuejs/eslint-plugin-vue/blob/master/tests/lib/rules/script-setup-uses-vars.js), I guess we can just delete [tests/lib/script-setup.js](https://github.com/vuejs/eslint-plugin-vue/blob/master/tests/lib/script-setup.js) then?

**Comment by ota-meshi** [comment]
As you say, `vue/script-setup-uses-vars` is rule for syntax (as an experimental feature) that was available early in Vue v3.0.
I think we can remove `tests/lib/script-setup.js`.

**Comment by FloEdelmann** [comment]
Alright, the test will be removed in #1881.
